src/container/cli.c

```diff
@@ -487,13 +487,93 @@ struct cont_open_args {
 	daos_handle_t		*hdlp;
 };
 
+struct pmap_refresh_cb_arg {
+	struct dc_pool		*pra_pool;
+	uint32_t		 pra_pm_ver;
+};
+
+static int
+pmap_refresh_cb(tse_task_t *task, void *data)
+{
+	struct pmap_refresh_cb_arg	*cb_arg;
+	struct dc_pool			*pool;
+	uint32_t			 pm_ver;
+	int				 rc = task->dt_result;
+
+	cb_arg = (struct pmap_refresh_cb_arg *)data;
+	pool = cb_arg->pra_pool;
+	if (rc)
+		goto out;
+
+	pm_ver = pool_map_get_version(pool->dp_map);
+	if (pm_ver < cb_arg->pra_pm_ver) {
+		/* very corner case, ask user to try-again-later */
+		rc = -DER_AGAIN;
+		D_ERROR(DF_UUID": pmap_refresh cannot get required "
+			"version (%d:%d), try again later "DF_RC"\n",
+			DP_UUID(pool->dp_pool), pm_ver,
+			cb_arg->pra_pm_ver, DP_RC(rc));
+	}
+out:
+	if (rc)
+		D_ERROR(DF_UUID": pmap_refresh(task %p) failed, "DF_RC"\n",
+			DP_UUID(pool->dp_pool), task, DP_RC(rc));
+	dc_pool_put(pool);
+	return rc;
+}
+
+static int
+pmap_refresh(tse_task_t *task, daos_handle_t poh, uint32_t pm_ver)
+{
+	struct dc_pool			*pool;
+	struct pmap_refresh_cb_arg	 cb_arg;
+	tse_sched_t			*sched;
+	daos_pool_query_t		*pargs;
+	tse_task_t			*ptask = NULL;
+	int				 rc;
+
+	pool = dc_hdl2pool(poh);
+	D_ASSERT(pool != NULL);
+	sched = (task != NULL) ? tse_task2sched(task) : NULL;
+	rc = dc_task_create(dc_pool_query, sched, NULL, &ptask);
+	if (rc != 0)
+		return rc;
```

Comment:
dc_pool_put()?

Reply:
Ah yes, L540 was "goto out", I changed it for some other reason but forgot this point.